### PR TITLE
Optimistic Minter

### PR DESCRIPTION
**Pull request by Joeysantoro**
https://tribe.fei.money/t/fip-xx-fei-minting-for-optimistic-approval/3565

### CODE REVIEW

File: contracts/token/minter/FeiTimedMinter.sol [hunk 1]
```diff
@@ -1,9 +1,9 @@
 pragma solidity ^0.8.0;
 
-import "../refs/CoreRef.sol";
-import "../utils/Timed.sol";
-import "../utils/Incentivized.sol";
-import "../utils/RateLimitedMinter.sol";
+import "../../refs/CoreRef.sol";
```
Comment (ditchfieldcaleb) [inline contracts/token/minter/FeiTimedMinter.sol:3]: If you're going to be changing the paths, you may as well use the @ notation.

File: contracts/token/minter/IPCVEquityMinter.sol [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 pragma solidity ^0.8.0;
 
-import "../oracle/collateralization/ICollateralizationOracle.sol";
+import "../../oracle/collateralization/ICollateralizationOracle.sol";
```
Comment (ditchfieldcaleb) [inline contracts/token/minter/IPCVEquityMinter.sol:5]: use @ paths

File: contracts/token/minter/PCVEquityMinter.sol [hunk 3]
```diff
@@ -2,8 +2,8 @@ pragma solidity ^0.8.0;
 
 import "./FeiTimedMinter.sol";
 import "./IPCVEquityMinter.sol";
-import "../Constants.sol";
-import "../pcv/IPCVSwapper.sol";
+import "../../Constants.sol";
```
Comment (ditchfieldcaleb) [inline contracts/token/minter/PCVEquityMinter.sol:5]: use @ paths

File: proposals/dao/fip_35.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { ethers } from 'hardhat';
+import chai, { expect } from 'chai';
+import CBN from 'chai-bn';
+import {
+  DeployUpgradeFunc,
+  NamedContracts,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '../../types/types';
```
Comment (ditchfieldcaleb) [inline proposals/dao/fip_35.ts:10]: use @ paths

File: proposals/dao/fip_35.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+chai.use(CBN(ethers.BigNumber));
+
+// Constants
+const TIMED_MINTER_FREQUENCY = '604800'; // weekly
```
Comment (ditchfieldcaleb) [inline proposals/dao/fip_35.ts:15]: Period, but w/e

**Review by ditchfieldcaleb — CHANGES_REQUESTED** [review]

**Review by ditchfieldcaleb — APPROVED** [review]

File: contracts/token/minter/OwnedTimedMinter.sol [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        uint256 _frequency,
+        uint256 _initialMintAmount
+    ) 
+        FeiTimedMinter(_core, _owner, 0, _frequency, _initialMintAmount)
+    {
+        transferOwnership(_owner);
+    }
+
+    /// @notice triggers a minting of FEI by owner
+    function mint() public override onlyOwner {
```
Comment (ElliotFriedman) [inline contracts/token/minter/OwnedTimedMinter.sol:29]: Who will the owner of this contract be? I would assume it's either the guardian or the OA multisig?

File: proposals/dao/fip_35.ts (hunk 5 above)
Comment (Joeysantoro) [reply proposals/dao/fip_35.ts:15]: Its frequency just do one round of mental gymnastics

**Review by ditchfieldcaleb — APPROVED** [review]

**Pull request merged**